### DISCUSSION TITLE
I'm considering to move from playing official servers to create my private server so I have some generic questions.

### DISCUSSION THREAD

**Discussion by SeravySensei** [Melledy/Grasscutter#2125] [General]
Ideally I'd like to turn Genshin into a game where I can freely go around in the open world, kill things, loot chests and like a "normal" RPG, get character materials, 5* artifacts etc from the chests and enemy drops instead of the resin system.
However I don't know how much is implemented on server side or client side in the game so no idea what is possible (assuming I can edit the source code and data to make it happen). I do have some previous experience from running my private Ragnarok Online server.

-How stable and bug free is the game currently? Is it worth jumping into playing it now, or wait another few months, or maybe years? I know private server development takes time but the official gameplay is getting more and more unbearable so I want to get away from it as soon as possible.
-I assume loot from chests and enemies is server side? Is it possible to define different drops for different enemy types or chests in each map region or is it the same loot for every chest and enemy of a given category? Can the drops include items not currently dropped by enemies or chests?
-I'd like to improve the artifact system, is it possible to change what main stats or substats an artifact can have, for example physical dmg as substat, a feather with pyro dmg or a goblet with energy recharge as main stat? Or change how much of each stat artifacts gain when upgraded to make it level all stats equally instead of at random?
-I'd also like a chest respawn mechanic, getting 1 random chest back whenever I open another once more than 40% of the ingame chests were opened, the chests do seem to be server side so I think this shouldn't be difficult to achieve?
-Is it possible to add new objects to the game map, such as new chests, new monster spawning points, or new waypoints?
-Are character movesets server or clientside? Can I adjust skill multipliers, cooldowns, energy generation, etc to make the "bad" characters more playable? For example make Dori produce more energy, burst for 60 and remove the self electro application?
-Can I add every single unit in the game to the starglitter shop and standard pool? I also would like a banner rotation where every past banner has a random chance to show up and they change every hour or so.
-Is it possible to have more maximal stamina and change electro and dendroculus to grant more?
-Is it possible the change Crystallize to "autoloot" the crystal and generate the shield immediately? How about autolooting items in general? Or changing Superconduct to debuff defense instead of physical resistance?
-Is it possible the change the movement speed of characters to be the same for everyone instead of slower for female characters?
-Is it possible to change artifact set effects (get rid of niche sets and add more universal ones such as a CR, CD reduction or ATK Speed set) or generate 5* versions of currently 4* only sets?
-Is it possible to add a way to enter quest locked areas, like an NPC that opens them? As far as I know quests are not implemented yet and honestly I wouldn't want to do them even if they were.

Finally, is it possible to play a private server while still occasionally playing on the official server (mostly to see new content)? Would that need two copies of the installed Genshin client? How often do people get banned for having a private server installed?

Oh and is there a good tutorial on how to install the client and server and compile the source? (and what's the best editor/compiler to use on windows?)

**Comment by NotThorny** [comment]
- Relatively stable but many things are just not implemented
- All drops from chests and mobs are fully customizable and can be anything you want
- Yes you can have artifacts be whatever you want and you could equally customize the upgrade system
- Maybe, but in general open world things aren't well implemented so respawning chests would be a full thing to implement
- Yes if you edit resource and data files (probably not waypoints, but you might be able to use the teapot ones in other scenes
- Avatar modification might be possible but I wouldn't count on it at all. Cooldowns can be modified with commands, energy recharge can be edited in the data file for it so yes you can make characters produce infinite energy every attack if you want.
- You can have anything you want in banners (even things that aren't avatars) and you can add whatever you want to wishing/shops/etc in data files
- Maybe but you'd need to look into src
- Autolooting is client modding/cheats not server function
- No but you can use the movespeed buff
- Maybe in resources files, but you can already fully customize what stats an artifact can have, which includes CDR (not as a set buff)
- Just teleport

Nobody has ever been banned for using private servers and you use the same client, having two copies is for people who are too lazy to copy/paste a single file.

Yes there's a tutorial right on the front page of this github, though I'd suggest using Cultivation instead -> https://github.com/Grasscutters/Cultivation
The setup section on the front of the github links to the youtube tutorial, or you can follow the written instruction right below it. 

You don't need any editor/compiler to build from source, gradlew.bat -> gradlew jar is all it takes to build (see bottom of readme)

**Comment by SeravySensei** [comment]
Thank you!
I think I'll give it a try as soon as I finish my other project and have the time needed for this.

**Reply by SeravySensei** [reply]
Installing isn't what I'm worried about, I still remember when I spent several hundred hours adjusting enemy drops and item stats on my Ragnarok server and that's the bare minimum before I will start playing :)

**Reply by SeravySensei** [reply]
Couldn't resist anymore and tried installing it :)
Got stuck on the "resource" folder. The wiki link gives a 404 and the main/src/resources folder doesn't seem to contain anything like what I need?
I've already installed Cultivation, Java, MongoDB and IntelliJ, forked the repository and downloaded it, haven't tried to modify or compile anything yet. Also downloaded the Jar and moved it to the Cultivation folder.

**Reply by SeravySensei** [reply]
Ok, will do if anything else happens, found the files a moment ago!
